common/compositor/va/varenderer.cpp

```diff
@@ -440,18 +448,33 @@ bool VARenderer::UpdateCaps() {
       if (use_default) {
         itr->second.value_ = itr->second.caps_.range.default_value;
       }
-      cbparam.type = VAProcFilterColorBalance;
-      cbparam.value = itr->second.value_;
-      cbparam.attrib = itr->second.caps_.type;
-      if (!cb_elements[static_cast<int>(itr->first)].CreateBuffer(
-              va_context_, VAProcFilterParameterBufferType,
-              sizeof(VAProcFilterParameterBufferColorBalance), 1, &cbparam)) {
-        return false;
-      }
-      filters_.push_back(cb_elements[static_cast<int>(itr->first)].buffer());
+      cbparam[index].type = VAProcFilterColorBalance;
+      cbparam[index].value = itr->second.value_;
+      cbparam[index].attrib = itr->second.caps_.type;
+      index++;
```

Comment:
Do we ever expect the line 440 **for** loop to not run at least once? Otherwise, won't index always be at least 1 and the **if** on line 460 always be ignored?